packages/vite-plugin-svelte/src/utils/sourcemap.ts

```diff
@@ -0,0 +1,44 @@
+import MagicString, { MagicStringOptions } from 'magic-string';
+import { diff_match_patch, DIFF_DELETE, DIFF_INSERT } from 'diff-match-patch';
+
+export function buildMagicString(
```

Comment:
Pretty neat utility, I wonder if there's an external library that does exactly this :D